torchao/prototype/low_bit_optim/adam.py

```diff
@@ -265,11 +266,28 @@ def __init__(
             bf16_stochastic_round=bf16_stochastic_round,
             is_adamw=False,
         )
+        self.dynamic_range_expansion = dynamic_range_expansion
 
     @staticmethod
-    def _subclass_zeros(p: Tensor, signed: bool, block_size: int):
-        return OptimStateFp8.zeros(p.shape, block_size, p.device)
+    def _subclass_zeros(p: Tensor, signed: bool, block_size: int, dynamic_range_expansion: bool):
+        return OptimStateFp8.zeros(p.shape, block_size, p.device, dynamic_range_expansion)
```

Comment:
You can make `_subclass_zeros()` a normal method instead of `staticmethod`, so that it can access `self.dynamic_range_expansion` directly -> don't need to pass in `dynamic_range_expansion` again. When I first wrote this, I didn't foresee such extension. If you want, you can remove `block_size` from all `_subclass_zeros()` signature too, since if we change it to a normal method, we can also access `self.block_size` directly.